services/provider/proto/provider.proto

```diff
@@ -118,13 +118,30 @@ message StorageConfigResponse{
     SystemAttributes systemAttributes = 3;
 }
 
+message RbdDriverRequirements {
+  bool generateOmapInfo = 1;
+}
+
+message CephFsDriverRequirements {
+}
+
+message NfsDriverRequirements {
+}
+
+message CsiDrivers {
+  RbdDriverRequirements DesiredRbdDriver = 1;
+  CephFsDriverRequirements DesiredCephFsDriver = 2;
+  NfsDriverRequirements DesiredNfsDriver = 3;
+}
+
 // GetDesiredClientStateResponse holds the response for the GetDesiredClientState API request
 message GetDesiredClientStateResponse{
   repeated bytes kubeResources = 1;
   string desiredStateHash = 2;
   string clientOperatorChannel = 3;
   bool maintenanceMode = 4;
   bool mirrorEnabled = 5;
+  CsiDrivers CsiDrivers = 6;
```

Comment:
Why do we need a substruct here? Can we just have the requirement sections on the root struct?